### PR TITLE
[extractor/ciscowebex] Support password-protected videos

### PR DESCRIPTION
**Pull request by damianoamatruda**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/ciscowebex.py [hunk 1]
```diff
@@ -38,11 +39,30 @@ def _real_extract(self, url):
...
+                if not password:
+                    raise ExtractorError(
+                        'This video is protected by a password, use the --video-password option', expected=True)
+                raise ExtractorError('Wrong password', expected=True)
+            raise ExtractorError(f'{self.IE_NAME} said: {stream["code"]} - {stream["message"]}', expected=True)
+
+        if urlh.status == 429:
+            self.raise_login_required(
+                f'{self.IE_NAME} asks you to solve a CAPTCHA. Solve CAPTCHA in browser and',
+                method=cookies)
```
Comment (pukkandan) [inline yt_dlp/extractor/ciscowebex.py:65]:
```suggestion
                method='cookies')
```

File: yt_dlp/extractor/ciscowebex.py [hunk 2]
```diff
@@ -72,13 +93,14 @@ def _real_extract(self, url):
...
                 })
+        self._sort_formats(formats)
 
         return {
             'id': video_id,
             'title': stream['recordName'],
             'description': stream.get('description'),
             'uploader': stream.get('ownerDisplayName'),
-            'uploader_id': stream.get('ownerUserName') or stream.get('ownerId'),  # mail or id
+            'uploader_id': stream.get('ownerUserName') or stream.get('ownerId'),
```
[the remark this answers is not in the document]
Comment (damianoamatruda) [reply yt_dlp/extractor/ciscowebex.py:101]: `ownerUserName` (e-mail address) _seems_ to be always available and is more informative, `ownerId` (ID corresponding to the e-mail address) is a backup value

File: yt_dlp/extractor/ciscowebex.py [hunk 3]
```diff
@@ -38,11 +39,30 @@ def _real_extract(self, url):
...
+        stream, urlh = self._download_json_handle(
             'https://%s.webex.com/webappng/api/v1/recordings/%s/stream' % (subdomain, video_id),
-            video_id, fatal=False, query={'siteurl': siteurl})
-        if not stream:
-            self.raise_login_required(method='cookies')
+            video_id, headers=headers, query={'siteurl': siteurl}, expected_status=(403, 429))
+
+        if urlh.status == 403:
+            if stream['code'] == 53004:
+                self.raise_login_required(method='cookies')
```
Comment (pukkandan) [inline yt_dlp/extractor/ciscowebex.py:54]:
```suggestion
                self.raise_login_required()
```

File: yt_dlp/extractor/ciscowebex.py [hunk 4]
```diff
@@ -38,11 +39,30 @@ def _real_extract(self, url):
...
+            if stream['code'] == 53005:
+                if not password:
+                    raise ExtractorError(
+                        'This video is protected by a password, use the --video-password option', expected=True)
+                raise ExtractorError('Wrong password', expected=True)
```
Comment (pukkandan) [inline yt_dlp/extractor/ciscowebex.py:59]:
```suggestion
                if password:
                    raise ExtractorError('Wrong password', expected=True)
                raise ExtractorError(
                    'This video is protected by a password, use the --video-password option', expected=True)
```

**Pull request merged**